frontend/src/HexView.svelte

```diff
@@ -197,91 +240,95 @@
   }
 </script>
 
-{#await dataPromise}
+{#await dataLenPromise}
   <LoadingAnimation />
-{:then dataResult}
-  {#if dataResult !== undefined && dataResult.byteLength > 0}
+{:then dataLen}
+  {#if dataLen > 0}
     <!-- 
       The magic number below is the largest height that Firefox will support with
       a position: sticky element. Otherwise, the sticky element scrolls away.
       Found this by manual binary search on my computer. 
     -->
     <div
       style:height="min(8940000px, calc(var(--line-height) * {Math.ceil(
-        dataResult.byteLength / alignment
+        dataLen / alignment
       )}))"
     >
       <div class="sticky">
         <div class="breadcrumb">
           <Breadcrumb />
         </div>
         <div class="hbox">
-          <div>
-            {#each chunks as _, chunkIndex}
-              <div>
-                {(chunkIndex * alignment + start)
-                  .toString(16)
-                  .padStart(8, "0") + ": "}
-              </div>
-            {/each}
-          </div>
-
-          <span class="spacer"> </span>
-
-          {#await childRangesPromise}
+          {#await chunkDataPromise}
+            <LoadingAnimation />
+          {:then chunks}
             <div>
-              {#each chunks as hexes}
+              {#each chunks as _, chunkIndex}
                 <div>
-                  {#each hexes as byte}
-                    <span class="byte">{byte}</span>
-                  {/each}
+                  {(chunkIndex * alignment + start)
+                    .toString(16)
+                    .padStart(8, "0") + ": "}
                 </div>
               {/each}
             </div>
-          {:then childRangesResult}
-            <div>
-              {#each chunks as hexes, chunkIndex}
-                <div>
-                  {#each hexes as byte, byteIndex}
-                    {@const rangeInfo = getRangeInfo(
-                      chunkIndex * alignment + byteIndex + start,
-                      childRangesResult,
-                      byte
-                    )}
-                    {#if rangeInfo?.resource_id === null || rangeInfo?.resource_id === undefined}
+
+            <span class="spacer"> </span>
+
+            {#await childRangesPromise}
```

Comment:
All these separate `#await` blocks should be combined into one with `Promise.all`. Then their values can be unpacked from the result. It's the equivalent of `asyncio.gather`, and doesn't resolve the promise until all of the promises passed as arguments resolve. That way there is only one loading screen, after which everything is good to go. 

There is a bit of a trade-off with this approach. At the moment, it loads data first, and only begins loading the ranges after the data has loaded. So it's quick to show something – the data – but slow to show the final view. Whereas the other method takes a bit longer to show the first thing – the data with highlighted ranges – but afterwards, the view is completely rendered.

Reply:
`dataChunkPromise` is consistently updated while scrolling, will it still await and update the result if i await this way?

Reply:
I am pretty sure it should, but it would be good to test a little and 100% confirm.

Reply:
It appears to reset the chunk data when scrolling, I had this issue as well when the `#await chunkDataPromise` block was too far up in the page. Idk if theres a way to get it to work with Promise.all

https://github.com/redballoonsecurity/ofrak/assets/14964383/9af722ff-c949-449b-a38f-37bd5da69fb6

Reply:
Damn, okay. Maybe collect all of the other promises except for the `chunkDataPromise` with a `Promise.all`? If it doesn't work, it's alright.

Reply:
I think the problem is the datalen promise, i could probably do childData and childRanges together though, I'll try that out.

Reply:
nevermind, this won't work because the result of chunkDataPromise is required for the "pre-await" block of childRanges.